io/include/traccc/io/read_geometry.hpp

```diff
@@ -32,11 +35,17 @@ template <typename detector_t>
 geometry alt_read_geometry(const detector_t& det) {
 
     const auto transforms = det.transform_store();
-    const auto surfaces = det.surfaces();
+    const auto surfaces =
```

Comment:
This will potentially only put the portals in the map. For all surfaces:
```suggestion
    const auto surfaces = det.surface_lookup();
```